File: nomad-samples/common/py/movies/create_movie/genre/create_genre.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def create_genre(AUTH_TOKEN, NAME, SLUG):
    if not AUTH_TOKEN:
        raise Exception("Authorization token not found")
  
    GET_API_URL = f"{ADMIN_API_URL}/content/new?contentDefinitionId={MOVIE_GENRE_CONTENT_DEFINITION_ID}"
    
    HEADERS = {
      "Content-Type": "application/json",
      "Authorization": "Bearer " + AUTH_TOKEN
    }
    try:
        NEW_RESPONSE = requests.get(GET_API_URL, headers=HEADERS)
    
        if not NEW_RESPONSE:
            raise Exception()
        
        INFO = NEW_RESPONSE.json()
        ID = INFO["contentId"]
    
    except:
        raise Exception()

    PUT_API_URL = f"{ADMIN_API_URL}/Content/{MOVIE_GENRE_CONTENT_DEFINITION_ID}"

    # Build the payload body
    BODY = {
        "contentId": ID,
        "contentDefinitionId": MOVIE_GENRE_CONTENT_DEFINITION_ID,
        "properties": {
            "name": NAME,
            "slug": SLUG,
        },
    }

    try:
        # Send POST request
        RESPONSE = requests.put(PUT_API_URL, headers=HEADERS, data=json.dumps(BODY))

        # Check for success
        if not RESPONSE.ok:
            raise Exception()
        
        # Get the response
        ID = RESPONSE.json()

        # Return the ID
        return ID
    except:
        api_exception_handler(RESPONSE, "Update genre failed")
    
    

File: nomad-samples/common/js/sync-1.1/genre/create-genre.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Create Genre
 *
 * @param {string} name         | The genre name
 * @param {string} slug         | The genre slug
 * @param {string} authToken    | The authorization token
 *
 * @returns {string} New genre ID
 */
export default async function createGenre(name, slug, authToken) {
    // Check for valid parameters
    if (!name || !slug || !authToken) {
        throw new Error("Create Genre: Invalid API call");
    }

    // Create header for the request
    const headers = new Headers();
    headers.append("Content-Type", "application/json");
    headers.append("Authorization", `Bearer ${authToken}`);

    // Build the payload body
    const body = {
        contentDefinitionId: prjConstants.MOVIE_GENRE_CONTENT_DEFINITION_ID,
        properties: {
            name: name,
            slug: slug
        }
    };

    // Send POST request
    const response = await fetch(`${prjConstants.SERVER_URL}/Content/${prjConstants.MOVIE_GENRE_CONTENT_DEFINITION_ID}`, {
        method: "POST",
        headers: headers,
        body: JSON.stringify(body)
    });

    // Check for success
    if (response && response.ok) {
        // Get the response
        const id = await response.text();

        // Return the ID
        return id.replaceAll('"', "");
    }

    await apiExceptionHandler(response, `Create Genre ${name} failed`);
}
